ddtrace/tracer/context_test.go

```diff
@@ -84,14 +84,12 @@ func TestStartSpanWithSpanLinks(t *testing.T) {
 	_, _, _, stop := startTestTracer(t)
 	defer stop()
 	spanLink := ddtrace.SpanLink{TraceID: 789, TraceIDHigh: 0, SpanID: 789, Attributes: map[string]string{"reason": "terminated_context", "context_headers": "datadog"}, Flags: 0}
-	var ctx ddtrace.SpanContext
-	ctx = &spanContext{spanID: 789, traceID: traceIDFrom64Bits(789), spanLinks: []ddtrace.SpanLink{spanLink}}
+	ctx := &spanContext{spanID: 789, traceID: traceIDFrom64Bits(789), spanLinks: []ddtrace.SpanLink{spanLink}}
 
 	t.Run("spanContext with spanLinks satisfies SpanContextWithLinks interface", func(t *testing.T) {
-		ctxLink, ok := ctx.(ddtrace.SpanContextWithLinks)
-		assert.True(t, ok)
-		assert.Equal(t, len(ctxLink.SpanLinks()), 1)
-		assert.Equal(t, ctxLink.SpanLinks()[0], spanLink)
+		var _ ddtrace.SpanContextWithLinks = ctx
```

Comment:
Should an assertion be made here to make sure that the casting doesn't fail? Otherwise what is the point of this declaration? 🤔

Reply:
The code won't compile if this fails.
https://dev.to/kittipat1413/checking-if-a-type-satisfies-an-interface-in-go-432n#:~:text=In%20Go%2C%20to%20check%20if,now%2C%20not%20at%20runtime.%E2%80%9D